torch/_inductor/codegen/cuda/cuda_scheduling.py

```diff
@@ -1,15 +1,205 @@
-from ... import config
+import logging
+from typing import Callable, cast, List
+
+from ... import config, ir
 from ...codecache import code_hash, get_path
+from ...ir import ComputedBuffer, CUDATemplateBuffer, Pointwise
+from ...scheduler import BaseSchedulerNode, FusedSchedulerNode, Scheduler, SchedulerNode
 from ...utils import get_fused_kernel_name, get_kernel_metadata
 from ...virtualized import V
-
 from ..common import IndentedBuffer
 from ..triton import TritonScheduling
 
+log = logging.getLogger(__name__)
+
+
+def _can_fuse_epilogue(
+    cuda_template_buffer: CUDATemplateBuffer,
+    epilogue_nodes: List[ir.Buffer],
+    additional_node: ir.Buffer,
```

Comment:
I'd expect this to be operating on SchedulerNode and FusedSchedulerNode objects?  Why is this running in ir.Buffers?

Reply:
This is a private function called by the Scheduler level code that operates on the unwrapped IR nodes. The fusion decision cannot be made without looking at these unwrapped IR nodes.